cat-loving-app/src/cats/RandomCatsScreen.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from 'react';
+import RandomCatsTop from '../common/RandomCatsTop';
+import CatImage from '../common/CatImage';
+import { getRandomCats, CatData } from '../services/catService';
+import { styled } from 'styled-components';
+
+
+const CatImageListContainer = styled.div`
+    position: relative;
+`;
+
+const CatImageContainer = styled.ul`
+    display: flex; 
+    flex-wrap: wrap;
+`;
+
+const RandomCatsScreen: React.FC = () => {
+    const [cats, setCats] = useState<CatData[]>([]);
+    const [imagesPerRow, setImagesPerRow] = useState(2);
+
+    useEffect(() => {
+        const savedCats = localStorage.getItem('cats');
+        if (savedCats) {
+            setCats(JSON.parse(savedCats));
+            return;
+        }
+    
+    const fetchData = async () => {
+        const cats = await getRandomCats();
+        setCats(cats);
+        localStorage.setItem('cats', JSON.stringify(cats));
+    };
+    fetchData();
+    }, []);
+
+    return (
+        <div>
+            <RandomCatsTop 
+                header="Random Cats"
+                imagesPerRow={imagesPerRow} 
+                setImagesPerRow={setImagesPerRow}  
+                getNewListOfCats={async ()=> {
+                    localStorage.removeItem('cats');
+                    const newCats = await getRandomCats();
+                    setCats(newCats);
+                    localStorage.setItem('cats', JSON.stringify(newCats));
+                }}
+            />
```

Comment:
Could you explain the logic of this component and why you choose to do it this way?

Reply:
I noticed this component does not follow common conventions, like the "getNewListOfCats", which appears to be an event ('onclick') action callback to a child component. It'd be expected to name it accordingly (eg. "onDisplayMoreCats").

Reply:
This component is the one that is the container of the screen that shows random Cats. The logic here is that the RandomCatsTop contains the ui that sets the number of images per row and gets a fresh list of cats. 

This is necessary since I store in localStorage every request for 10 random cats. I store the requests in localStorage in order to make more seamless the experience of a user opening the CatDetail inside the App and not in a modal. In that case when the user was hitting back he/she would get a new set of 10 cats. For that reason I have implemented the button getNewListOfCats in order to clean your local storage

I am totally in agreement that it should be named with a name starting with 'on' to indicate the action

Reply:
I see. In that case, you think using localStorage is better than sessionStorage? 

How would you clear the history of loaded cat images for the user to start all anew when they reopen the app? Assuming this was an actual spec/requirement.

Reply:
I think you are right. SessionStorage should be better as long there's no strict requirement needing to persist data even after the tab is closed. 

I could use `localStorage.removeItem(key);` This looks implicit and it only removes the specific item.